porting/npl/linux/src/wqueue.h

```diff
@@ -0,0 +1,76 @@
+/*
+   wqueue.h
+   Worker thread queue based on the Standard C++ library list
+   template class.
+   ------------------------------------------
+   Copyright (c) 2013 Vic Hargrave
+   Licensed under the Apache License, Version 2.0 (the "License");
+   you may not use this file except in compliance with the License.
+   You may obtain a copy of the License at
+       http://www.apache.org/licenses/LICENSE-2.0
+   Unless required by applicable law or agreed to in writing, software
+   distributed under the License is distributed on an "AS IS" BASIS,
+   WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+   See the License for the specific language governing permissions and
+   limitations under the License.
+*/
+
+// https://vichargrave.github.io/articles/2013-01/multithreaded-work-queue-in-cpp
+// https://github.com/vichargrave/wqueue/blob/master/wqueue.h
+
+
+#ifndef __wqueue_h__
+#define __wqueue_h__
+
+#include <pthread.h>
+#include <list>
+
+using namespace std;
+
+template <typename T> class wqueue
```

Comment:
do not mix C++ into C code - this has to be rewritten

we can have C++ port for something, but this should be separate NPL

Reply:
Why require the busy work to rewrite this working, tested C++ queue implementation?
It would only be used by the Linux NPL porting layer.

Reply:
because this is C project and it should build with C compiler.
as I said, we can add C++ linux port, but in this case I'd expect it also uses C++ standard library for threads, mutexes etc.

Reply:
@andrzej-kaczmarek Is removing the C++ the last item blocking acceptance of this PR?

Reply:
IIRC there were some more comments to be fixed, like macros defined in NPL instead of common code (perhaps some more, I do not remember now)